File: Math/204_CountPrimes_TLE.java
public class Solution {
    public int countPrimes(int n) {
        if (n < 2) {
            return 0;
        }
        boolean[] tbl = new boolean[n];
        tbl[0] = tbl[1] = true;
        for (int j = 3; j < n; j++) {
            if (j % 2 == 0) tbl[j] = true;
        }

        for (int i=3; i*i < n; i++) {
            if(!tbl[i]) {
                for (int j=i+1; j<n; j++) {
                    if(!tbl[i] && j % i == 0) tbl[j] = true;
                }
            }
        }

        int cnt = 0;
        for (int i=2; i<n; i++) {
            if(!tbl[i]) cnt++;
        }
        return cnt;
    }
}

File: Math/204_CountPrimes_TLE.py
class Solution(object):

    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """
        if n < 2:
            return 0
        table = [False] * n
        for i in range(3, n):
            if i % 2 == 0:
                table[i] = True
        for i in range(3, n):
            if i * i > n:
                break
            if not table[i]:
                for j in range(i + 1, n):
                    if j % i == 0:
                        table[j] = True
        cnt = 0
        for x in range(2, n):
            if not table[x]:
                cnt += 1
        return cnt

if __name__ == '__main__':
    s = Solution()
    # if using this test case, it will be TLE
    print s.countPrimes(499979)
